src/collector/layer-zero/layer-zero.worker.ts

```diff
@@ -0,0 +1,617 @@
+     * The function ensures proper handling and formatting of Ethereum addresses and numeric IDs.
+     * The first 2 bytes of the encoded header are skipped as they represent the version, later 
+     * instead of using a counter to skip bytes, the slice function is used to extract the required.
+     * 
+     * @param encodedHeader - The encoded header string to be decoded.
+     * @returns An object containing the decoded header fields.
+     */
+    private decodeHeader(encodedHeader: string): any {
+        const version = encodedHeader.slice(2, 2 + 2);
+        const nonce = encodedHeader.slice(2 + 2, 2 + 2 + 16);
+        const srcEid = Number('0x' + encodedHeader.slice(2 + 18, 2 + 18 + 8));
+        const sender = '0x' + encodedHeader.slice(2 + 26, 2 + 26 + 64).slice(24);
+        const dstEid = Number('0x' + encodedHeader.slice(2 + 90, 2 + 90 + 8));
+        const receiver = '0x' + encodedHeader.slice(2 + 98, 2 + 98 + 64).slice(24);
+
+        return {
+            version,
+            nonce: Number('0x' + nonce),
+            srcEid,
+            sender,
+            dstEid,
+            receiver,
+        };
+    }
+
+
+    private calculatePayloadHash(guid: string, message: string): string {
+        const payload = `0x${guid}${message}`;
+        return ethers.keccak256(payload);
+    }
+}
+
+/**
+     * Checks if the configuration is verifiable.
+     * 
+     * @param receiveULN302 - The ULN302 contract instance.
+     * @param config - The ULN configuration.
+     * @param headerHash - The header hash.
+     * @param payloadHash - The payload hash.
+     * @returns A boolean indicating if the configuration is verifiable.
+     */
+async function checkIfVerifiable(
+    receiveULN302: ReceiveULN302,
+    config: UlnConfigStruct,
+    headerHash: BytesLike,
+    payloadHash: BytesLike,
+): Promise<boolean> {
+    try {
+        const requiredDVNs = config.requiredDVNs.map(dvn => dvn.toString());
+        const optionalDVNs = config.optionalDVNs.map(dvn => dvn.toString());
+        const formatConfig: UlnConfigStruct = {
+            confirmations: '0x' + config.confirmations.toString(16).padStart(16, '0'),
+            requiredDVNCount:
+                '0x' + config.requiredDVNCount.toString(16).padStart(2, '0'),
+            optionalDVNCount:
+                '0x' + config.optionalDVNCount.toString(16).padStart(2, '0'),
+            optionalDVNThreshold:
+                '0x' + config.optionalDVNThreshold.toString(16).padStart(2, '0'),
+            requiredDVNs: requiredDVNs,
+            optionalDVNs: optionalDVNs,
+        };
+        const isVerifiable = await receiveULN302.verifiable(
+            formatConfig,
+            headerHash,
+            payloadHash,
+        );
+        return isVerifiable;
+    } catch (error) {
+        console.error('Error verifying the configuration: ', error);
+        throw new Error('Error verifying the configuration: error details.');
```

Comment:
Remove the `console.error` and update the error statement. This is not ideal, as the `verifiable()` call is only given one chance to succeed, but let's move forward for now.
```suggestion
        throw new Error('Error verifying the payload.');
```

Reply:
Check on this if I solved as expected